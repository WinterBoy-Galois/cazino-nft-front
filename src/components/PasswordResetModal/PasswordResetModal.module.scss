@import '../../styles/shared';

.error {
  margin-bottom: 1rem;
}

.container {
  margin: 3rem 0 0 0;

  @include media-breakpoint-up(md) {
    margin: 5rem auto;
  }
}

.button {
  @include media-breakpoint-up(md) {
    width: 70%;
    margin: 0 auto;
  }
}
