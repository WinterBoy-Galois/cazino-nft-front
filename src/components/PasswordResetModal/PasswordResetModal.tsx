import React, { useState, useEffect } from 'react';
import styles from './PasswordResetModal.module.scss';
import Modal, { transitionTimeout } from '../Modal';
import { useMutation } from '@apollo/client';
import { RESET_PASSWORD } from '../../graphql/mutations';
import { useStateValue } from '../../state';
import { success } from '../Toast';
import ApplicationError from '../../models/applicationError.model';
import { useTranslation } from 'react-i18next';
import { getFromGraphQLErrors, getFromGenericErrors } from '../../common/util/error.util';
import ErrorSummary from '../ErrorSummary';
import PasswordInput from '../PasswordInput';
import { useFormik } from 'formik';
import { validationSchema } from './lib/validationSchema';
import SpinnerButton from '../SpinnerButton';
import { useQueryParams } from '../../hooks/useQueryParams.hook';
import { useLocation, useNavigate } from '@reach/router';

interface IProps {
  show: boolean;
  loading: boolean;
  errors?: ApplicationError[];
  onClose?: () => void;
  onPasswordReset?: (newPassword: string) => void;
}

const PasswordResetModal: React.FC<IProps> = ({
  show,
  loading,
  errors,
  onClose,
  onPasswordReset = () => null,
}) => {
  const { t } = useTranslation(['auth']);

  const formik = useFormik({
    initialValues: {
      password: '',
      confirmPassword: '',
    },
    validateOnMount: true,
    validationSchema: validationSchema(t),
    onSubmit: async values => {
      onPasswordReset(values.password);
    },
  });

  const handleClose = () => {
    setTimeout(() => {
      formik.resetForm({});
      formik.validateForm();
    }, transitionTimeout);

    if (onClose) {
      onClose();
    }
  };

  return (
    <Modal show={show} onClose={handleClose} title={t('passwordReset.headline')}>
      <div className="row">
        <div className={`col-12 col-md-8 ${styles.container}`}>
          {errors && (
            <ErrorSummary errors={errors} showGeneralErrorsOnly={false} className={styles.error} />
          )}

          <form onSubmit={formik.handleSubmit}>
            <PasswordInput
              label={t('labels.password')}
              name="password"
              onChangeValue={v => formik.setFieldValue('password', v)}
              onBlur={formik.handleBlur}
              value={formik.values.password}
              {...(formik.touched.password ? { validationMessage: formik.errors.password } : {})}
            />

            <PasswordInput
              label={t('labels.confirmPassword')}
              name="confirmPassword"
              onChangeValue={v => formik.setFieldValue('confirmPassword', v)}
              onBlur={formik.handleBlur}
              value={formik.values.confirmPassword}
              {...(formik.touched.confirmPassword
                ? { validationMessage: formik.errors.confirmPassword }
                : {})}
            />

            <SpinnerButton
              type="submit"
              color="SECONDARY"
              loading={loading}
              className={styles.button}
            >
              {t('passwordReset.button')}
            </SpinnerButton>
          </form>
        </div>
      </div>
    </Modal>
  );
};

interface IWithDataProps {
  show: boolean;
  onClose?: () => void;
}

const PasswordResetModalWithData: React.FC<IWithDataProps> = ({
  show,
  onClose,
}: IWithDataProps) => {
  const { t } = useTranslation(['auth', 'common']);
  const [resetPassword, { loading }] = useMutation(RESET_PASSWORD);
  const params = useQueryParams();
  const [errors, setErrors] = useState<ApplicationError[]>();
  const [
    {
      auth: { passwordResetToken: token, state },
    },
    dispatch,
  ] = useStateValue();
  const location = useLocation();
  const navigate = useNavigate();

  const onPasswordReset = async (newPassword: string) => {
    const { data, errors: resetPasswordErrors } = await resetPassword({
      variables: { token, newPassword },
    });

    if (resetPasswordErrors) {
      return setErrors(getFromGraphQLErrors(resetPasswordErrors, t));
    } else if (data?.resetPassword?.errors) {
      return setErrors(getFromGenericErrors(data.resetPassword.errors, t));
    }

    dispatch({ type: 'AUTH_SIGN_IN', payload: data.resetPassword });

    navigate(location.pathname);
    success(t('passwordReset.success'));
  };

  const handleClose = () => {
    if (onClose) {
      onClose();
    }

    setErrors([]);
  };

  useEffect(() => {
    if (params?.token) {
      dispatch({ type: 'AUTH_ADD_PASSWORD_RESET_TOKEN', payload: params.token });
    }
  }, [params, dispatch]);

  if (show && (state === 'SIGNED_IN' || !params?.token)) {
    navigate(location.pathname);
    return null;
  }

  return (
    <PasswordResetModal
      show={show}
      loading={loading}
      errors={errors}
      onClose={handleClose}
      onPasswordReset={onPasswordReset}
    />
  );
};

export default PasswordResetModal;
export { PasswordResetModalWithData };
