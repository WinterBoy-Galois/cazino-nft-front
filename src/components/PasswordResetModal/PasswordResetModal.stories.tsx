import React from 'react';
import { boolean } from '@storybook/addon-knobs';

import PasswordResetModal from '.';
import { action } from '@storybook/addon-actions';
import { withA11y } from '@storybook/addon-a11y';

export default {
  title: 'Components/PasswordResetModal',
  component: PasswordResetModal,
  decorators: [withA11y],
};

const data = {
  errors: [
    {
      code: 'UUID_EXPIRED',
      message: 'Activation code has been expired',
    },
  ],
};

export const Default = () => (
  <PasswordResetModal
    loading={boolean('Loading', false)}
    show={boolean('Show', true)}
    onPasswordReset={action('onPasswordReset')}
  />
);

export const withError = () => (
  <PasswordResetModal
    loading={boolean('Loading', false)}
    show={boolean('Show', true)}
    onPasswordReset={action('onPasswordReset')}
    errors={data.errors}
  />
);
