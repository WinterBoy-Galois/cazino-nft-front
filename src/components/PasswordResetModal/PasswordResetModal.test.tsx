import React from 'react';
import { render } from '@testing-library/react';

import PasswordResetModal from '.';

describe('PasswordResetModal', () => {
  it('should match snapshot', async () => {
    // Arrange

    // Act
    const container = render(<PasswordResetModal show loading={false} />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
