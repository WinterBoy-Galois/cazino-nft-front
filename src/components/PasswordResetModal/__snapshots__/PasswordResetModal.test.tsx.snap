// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordResetModal should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div>
      <div
        class="container"
        style="height: 768px;"
      >
        <div
          aria-modal="true"
          class="modal scale--enter scale--enter-active"
          role="dialog"
        >
          <div
            class="modal__header"
          >
            <div
              class="modal__header__headline"
            >
              password reset
            </div>
            <div
              class="modal__header__close"
            >
              <svg
                class="icon icon__close"
                viewBox="0 0 365.696 365.696"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M243.188 182.86L356.32 69.726c12.5-12.5 12.5-32.766 0-45.247L341.238 9.398c-12.504-12.503-32.77-12.503-45.25 0L182.86 122.528 69.727 9.374c-12.5-12.5-32.766-12.5-45.247 0L9.375 24.457c-12.5 12.504-12.5 32.77 0 45.25l113.152 113.152L9.398 295.99c-12.503 12.503-12.503 32.769 0 45.25L24.48 356.32c12.5 12.5 32.766 12.5 45.247 0l113.132-113.132L295.99 356.32c12.503 12.5 32.769 12.5 45.25 0l15.081-15.082c12.5-12.504 12.5-32.77 0-45.25zm0 0"
                />
              </svg>
            </div>
          </div>
          <div
            class="modal__main"
          >
            <div
              class="row"
            >
              <div
                class="col-12 col-md-8 container"
              >
                <form>
                  <div
                    class="inputField__container"
                  >
                    <div
                      class="inputField__wrapper"
                    >
                      <div
                        class="inputField__group"
                      >
                        <label
                          class="inputField__label"
                        >
                          Password
                        </label>
                        <input
                          autocomplete="off"
                          class="inputField"
                          name="password"
                          type="password"
                          value=""
                        />
                      </div>
                      <button
                        class="inputField__icon"
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          class="fill inputField__icon--invisible"
                          viewBox="0 0 28.49 21.53"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            data-name="Layer 2"
                            id="Layer_2"
                          >
                            <g
                              data-name="Layer 1"
                              id="Layer_1-2"
                            >
                              <path
                                d="M28.33 10.1a22.8 22.8 0 00-6.44-7.51l-1.72 2.47a19.79 19.79 0 015.1 5.71c-1.2 2-5.15 7.76-11 7.76a9.26 9.26 0 01-3.08-.53l-1.76 2.54a12.26 12.26 0 004.81 1c8.94 0 13.88-9.68 14.09-10.09a1.5 1.5 0 000-1.35zM8.31 16.47a19.72 19.72 0 01-5.09-5.7c1.2-2 5.15-7.77 11-7.77a9.27 9.27 0 013.06.53L19.06 1a12.31 12.31 0 00-4.82-1C5.31 0 .36 9.68.16 10.1a1.5 1.5 0 000 1.34 22.64 22.64 0 006.44 7.5z"
                              />
                              <path
                                d="M12.27 10.77a2 2 0 011.28-1.85l2-2.92a4.88 4.88 0 00-1.34-.21 5 5 0 00-4 7.9zM16.22 10.76a2 2 0 01-1.29 1.84l-2 2.92a4.63 4.63 0 001.34.21 5 5 0 005-5 5 5 0 00-1-2.93zM7.48 21.38a1 1 0 01-.48-.16.94.94 0 01-.23-1.3L20.27.45a.92.92 0 011.29-.24.93.93 0 01.23 1.3L8.25 21a.93.93 0 01-.77.38z"
                              />
                            </g>
                          </g>
                        </svg>
                      </button>
                    </div>
                  </div>
                  <div
                    class="inputField__container"
                  >
                    <div
                      class="inputField__wrapper"
                    >
                      <div
                        class="inputField__group"
                      >
                        <label
                          class="inputField__label"
                        >
                          Confirm Password
                        </label>
                        <input
                          autocomplete="off"
                          class="inputField"
                          name="confirmPassword"
                          type="password"
                          value=""
                        />
                      </div>
                      <button
                        class="inputField__icon"
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          class="fill inputField__icon--invisible"
                          viewBox="0 0 28.49 21.53"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            data-name="Layer 2"
                            id="Layer_2"
                          >
                            <g
                              data-name="Layer 1"
                              id="Layer_1-2"
                            >
                              <path
                                d="M28.33 10.1a22.8 22.8 0 00-6.44-7.51l-1.72 2.47a19.79 19.79 0 015.1 5.71c-1.2 2-5.15 7.76-11 7.76a9.26 9.26 0 01-3.08-.53l-1.76 2.54a12.26 12.26 0 004.81 1c8.94 0 13.88-9.68 14.09-10.09a1.5 1.5 0 000-1.35zM8.31 16.47a19.72 19.72 0 01-5.09-5.7c1.2-2 5.15-7.77 11-7.77a9.27 9.27 0 013.06.53L19.06 1a12.31 12.31 0 00-4.82-1C5.31 0 .36 9.68.16 10.1a1.5 1.5 0 000 1.34 22.64 22.64 0 006.44 7.5z"
                              />
                              <path
                                d="M12.27 10.77a2 2 0 011.28-1.85l2-2.92a4.88 4.88 0 00-1.34-.21 5 5 0 00-4 7.9zM16.22 10.76a2 2 0 01-1.29 1.84l-2 2.92a4.63 4.63 0 001.34.21 5 5 0 005-5 5 5 0 00-1-2.93zM7.48 21.38a1 1 0 01-.48-.16.94.94 0 01-.23-1.3L20.27.45a.92.92 0 011.29-.24.93.93 0 01.23 1.3L8.25 21a.93.93 0 01-.77.38z"
                              />
                            </g>
                          </g>
                        </svg>
                      </button>
                    </div>
                  </div>
                  <button
                    class="button button secondary button button--small"
                    color="SECONDARY"
                    type="submit"
                  >
                    reset password
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      style="height: 768px;"
    >
      <div
        aria-modal="true"
        class="modal scale--enter scale--enter-active"
        role="dialog"
      >
        <div
          class="modal__header"
        >
          <div
            class="modal__header__headline"
          >
            password reset
          </div>
          <div
            class="modal__header__close"
          >
            <svg
              class="icon icon__close"
              viewBox="0 0 365.696 365.696"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M243.188 182.86L356.32 69.726c12.5-12.5 12.5-32.766 0-45.247L341.238 9.398c-12.504-12.503-32.77-12.503-45.25 0L182.86 122.528 69.727 9.374c-12.5-12.5-32.766-12.5-45.247 0L9.375 24.457c-12.5 12.504-12.5 32.77 0 45.25l113.152 113.152L9.398 295.99c-12.503 12.503-12.503 32.769 0 45.25L24.48 356.32c12.5 12.5 32.766 12.5 45.247 0l113.132-113.132L295.99 356.32c12.503 12.5 32.769 12.5 45.25 0l15.081-15.082c12.5-12.504 12.5-32.77 0-45.25zm0 0"
              />
            </svg>
          </div>
        </div>
        <div
          class="modal__main"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-8 container"
            >
              <form>
                <div
                  class="inputField__container"
                >
                  <div
                    class="inputField__wrapper"
                  >
                    <div
                      class="inputField__group"
                    >
                      <label
                        class="inputField__label"
                      >
                        Password
                      </label>
                      <input
                        autocomplete="off"
                        class="inputField"
                        name="password"
                        type="password"
                        value=""
                      />
                    </div>
                    <button
                      class="inputField__icon"
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        class="fill inputField__icon--invisible"
                        viewBox="0 0 28.49 21.53"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          data-name="Layer 2"
                          id="Layer_2"
                        >
                          <g
                            data-name="Layer 1"
                            id="Layer_1-2"
                          >
                            <path
                              d="M28.33 10.1a22.8 22.8 0 00-6.44-7.51l-1.72 2.47a19.79 19.79 0 015.1 5.71c-1.2 2-5.15 7.76-11 7.76a9.26 9.26 0 01-3.08-.53l-1.76 2.54a12.26 12.26 0 004.81 1c8.94 0 13.88-9.68 14.09-10.09a1.5 1.5 0 000-1.35zM8.31 16.47a19.72 19.72 0 01-5.09-5.7c1.2-2 5.15-7.77 11-7.77a9.27 9.27 0 013.06.53L19.06 1a12.31 12.31 0 00-4.82-1C5.31 0 .36 9.68.16 10.1a1.5 1.5 0 000 1.34 22.64 22.64 0 006.44 7.5z"
                            />
                            <path
                              d="M12.27 10.77a2 2 0 011.28-1.85l2-2.92a4.88 4.88 0 00-1.34-.21 5 5 0 00-4 7.9zM16.22 10.76a2 2 0 01-1.29 1.84l-2 2.92a4.63 4.63 0 001.34.21 5 5 0 005-5 5 5 0 00-1-2.93zM7.48 21.38a1 1 0 01-.48-.16.94.94 0 01-.23-1.3L20.27.45a.92.92 0 011.29-.24.93.93 0 01.23 1.3L8.25 21a.93.93 0 01-.77.38z"
                            />
                          </g>
                        </g>
                      </svg>
                    </button>
                  </div>
                </div>
                <div
                  class="inputField__container"
                >
                  <div
                    class="inputField__wrapper"
                  >
                    <div
                      class="inputField__group"
                    >
                      <label
                        class="inputField__label"
                      >
                        Confirm Password
                      </label>
                      <input
                        autocomplete="off"
                        class="inputField"
                        name="confirmPassword"
                        type="password"
                        value=""
                      />
                    </div>
                    <button
                      class="inputField__icon"
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        class="fill inputField__icon--invisible"
                        viewBox="0 0 28.49 21.53"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          data-name="Layer 2"
                          id="Layer_2"
                        >
                          <g
                            data-name="Layer 1"
                            id="Layer_1-2"
                          >
                            <path
                              d="M28.33 10.1a22.8 22.8 0 00-6.44-7.51l-1.72 2.47a19.79 19.79 0 015.1 5.71c-1.2 2-5.15 7.76-11 7.76a9.26 9.26 0 01-3.08-.53l-1.76 2.54a12.26 12.26 0 004.81 1c8.94 0 13.88-9.68 14.09-10.09a1.5 1.5 0 000-1.35zM8.31 16.47a19.72 19.72 0 01-5.09-5.7c1.2-2 5.15-7.77 11-7.77a9.27 9.27 0 013.06.53L19.06 1a12.31 12.31 0 00-4.82-1C5.31 0 .36 9.68.16 10.1a1.5 1.5 0 000 1.34 22.64 22.64 0 006.44 7.5z"
                            />
                            <path
                              d="M12.27 10.77a2 2 0 011.28-1.85l2-2.92a4.88 4.88 0 00-1.34-.21 5 5 0 00-4 7.9zM16.22 10.76a2 2 0 01-1.29 1.84l-2 2.92a4.63 4.63 0 001.34.21 5 5 0 005-5 5 5 0 00-1-2.93zM7.48 21.38a1 1 0 01-.48-.16.94.94 0 01-.23-1.3L20.27.45a.92.92 0 011.29-.24.93.93 0 01.23 1.3L8.25 21a.93.93 0 01-.77.38z"
                            />
                          </g>
                        </g>
                      </svg>
                    </button>
                  </div>
                </div>
                <button
                  class="button button secondary button button--small"
                  color="SECONDARY"
                  type="submit"
                >
                  reset password
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
